pkgs/pub_formats/tool/generate.dart

```diff
@@ -33,67 +33,70 @@ void main(List<String> args) {
   var generatedCount = 0;
   var changedCount = 0;
 
-  final schemaFile = File.fromUri(
-    packageRoot.resolve('doc/schema/pubspec_lock.schema.json'),
-  );
-  final schemaJson = jsonDecode(schemaFile.readAsStringSync());
-  final schema = JsonSchema.create(schemaJson);
+  for (final name in ['pubspec_lock', 'pubspec']) {
+    final schemaFile = File.fromUri(
+      packageRoot.resolve('doc/schema/${name}.schema.json'),
+    );
+    final schemaJson = jsonDecode(schemaFile.readAsStringSync());
+    final schema = JsonSchema.create(schemaJson);
 
-  final analyzedSchema = SchemaAnalyzer(
-    schema,
-    nameOverrides: {'path': 'path\$'},
-  ).analyze();
-  final textDumpFile = File.fromUri(
-    packageRoot.resolve('lib/src/pubspec_lock_syntax.g.txt'),
-  );
-  final newTextDumpContent = analyzedSchema.toString();
-  if (dumpAnalyzedSchema) {
-    var oldTextDumpContent = '';
-    if (textDumpFile.existsSync()) {
-      oldTextDumpContent = textDumpFile.readAsStringSync();
-    }
-    if (oldTextDumpContent != newTextDumpContent) {
-      textDumpFile.writeAsStringSync(newTextDumpContent);
-      print('Generated ${textDumpFile.uri}');
+    final analyzedSchema = SchemaAnalyzer(
+      schema,
+      nameOverrides: {'path': 'path\$'},
+      publicSetters: ['PubspecYamlFile'],
```

Comment:
What does this do? What does this do? (I couldn't find that class name in the generated files)

Reply:
Ah, I see. Would be helpful to put some of that explanation in the docs for "publicSetters" :)